### ISSUE TITLE
Red-pill me on SEO

### ISSUE THREAD

**Issue by LukeSmithxyz** [LukeSmithxyz/landchad#23]
Red-pill me on Search Engine Optimization.

While I usually don't care about that kind of stuff, It hit me that it might actually be a good idea to try to optimize these articles' chances of appearing in relevant searches.

Any obvious things we can standardize to increase visibility? Some kind of extra metadata? I legit know very little about SEO.

**Comment by zacharytyhacz** [comment]
For good seo, just meta tags in your `<head>`, and good use of `<h1>` and the other header tags.

Web scrapers will pull a site's HTML and use "standard" ( "standards" made by twitter + facebook )  selectors to find in that HTML to pull the best general info about your site quickly. Like youtube tags, but with how your title + thumbnail and other meta data are also taken into consideration.

A lot of the meta tags are tied in with twitter + facebook, which search engines and other sites quickly grab to show an image, title, description, etc even if you are not associated with those sites.


Here's an example of one of my sites that has full meta tags:

```
<head>
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0, shrink-to-fit=no">

	// this is for web scrapers like duckduckgo, google, etc
	<meta name="robots" content="index, follow">

	// self explanatory
	<meta name="description" content="A beautiful place where reading meets college sports">
	<meta name="keywords" content="reading, college, sports, college sports, nc state, north carolina, book, books, lessons for, lesson, education, children">
	<meta name="author" content="Luke Smith">

	// og: stuff is facebook stuff, but still many scrapers look for it since it's basically a standard at this point.
	<meta property="og:site_name" content="Site name" >
	<meta name="twitter:domain" property="twitter:domain" content="landchat.net">
	<meta name="og:title" property="og:title" content="Site title" >
	<meta property="og:description" content="A beautiful place where reading meets college sports" >
	<meta name="twitter:description" property="twitter:description" content="A beautiful place where reading meets college sports">
	<meta property="og:image" content="https://direct-image-link/images/tw-fb-li-card.png" >

	// if people share your content on twitter, these images will format their url and look cool on twitter
	<meta property="twitter:card" content="https://direct-image-link/images/tw-fb-li-card.png">
	<meta name="twitter:image:src" property="twitter:image:src" content="https://direct-image-link/images/tw-fb-li-card.png">
	<meta name="twitter:image" property="twitter:image" content="https://direct-image-link/images/tw-fb-li-card.png">
	<meta name="og:image:alt" property="og:image:alt" content="like img alt text" >

        // self explanatory 
	<meta property="og:url" content="https://landchad.net" >
	<meta property="og:type" content="website">


	// if you have accounts on twitter + facebook to tie your site's relevancy to, and vice verca
	<meta property="fb:admins" content="if you have a facebook group lol" >
	<meta name="twitter:site" property="twitter:site" content="@your twitter handler">
<meta name="twitter:creator" property="twitter:creator" content="@your twitter handler">
</head>
```

**Comment by tenseoverflow** [comment]
You can also use lighthouse from your chromium browser or from [here (https://web.dev)](https://web.dev/measure/) to analyse your site's SEO, accessibility or performance.

Google has its [search console thing](https://search.google.com/search-console/about), might also interest you.

**Comment by oidz1234** [comment]
Sites linking to landchad.net are also important, so we all link to landchad from our personal websites that would legitimately boost the rankings. (I think is called back linking in SEO terms)

**Comment by termermc** [comment]
External links are the biggest thing to help SEO. If you have a lot of sites linking to your site, you will be higher up. But meta tags are also very useful. With meta images, you also have a fun easter egg where you can put images that will pop up if people post a link to your site on social media, discord, etc, since you won't see it on the browser, but the images will show up on embeds on those types of sites and services. I do that for articles on my site.

**Comment by Cloud11665** [comment]
+ when somebody mentions your site on soycial-media, when you don't have the opengraph meta tags, it will just be a small hyperlink, but when you have your `og:image` and others set, it will display a big bold image of your choice, which (in this day and age of short attention spans) will bring more people into the path for the decentralization of the interwebz.

**Comment by phirecc** [comment]
> You can also use lighthouse from your chromium browser or from [here (https://web.dev)](https://web.dev/measure/) to analyse your site's SEO, accessibility or performance.

Did a bit of work on my site to give it a score of 100 everywhere (https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fphire.cc%2FOffline-Wikipedia-on-the-Kobo.html)
Mostly:

- Have images as lossless webp with png alternatives in picture tags (webp files are much smaller than png)
- Add meta-tag descriptions to articles
- Set a high value Cache-Control header for static files (images, fonts)
- Add alt tags to images

**Comment by zacharytyhacz** [comment]
> Add meta-tag descriptions to articles

Yes, we need to make sure contributors do this. For example, for the nginx article it must have relevant meta tag information for that specific article so the article as itself can be found via search engines, not just the entire site/home page.

**Comment by DGKSK8LIFE** [comment]
> > Add meta-tag descriptions to articles
> 
> Yes, we need to make sure contributors do this. For example, for the nginx article it must have relevant meta tag information for that specific article so the article as itself can be found via search engines, not just the entire site/home page.

This could be automated with [DOM](https://www.w3schools.com/jsref/dom_obj_meta.asp).

**Comment by ccuser44** [comment]
robots.txt

**Comment by binex-dsk** [comment]
> > > Add meta-tag descriptions to articles
> > 
> > 
> > Yes, we need to make sure contributors do this. For example, for the nginx article it must have relevant meta tag information for that specific article so the article as itself can be found via search engines, not just the entire site/home page.
> 
> This could be automated with [DOM](https://www.w3schools.com/jsref/dom_obj_meta.asp).

this man has really suggested using javascript

meta tags, images, external links, as said here, should be enough. alternatively pay Jewgle a very large sum of money

**Comment by thedavidhk** [comment]
> For good seo, just meta tags in your `<head>`, and good use of `<h1>` and the other header tags.
> 
> ...
> 	<meta name="keywords" content="reading, college, sports, college sports, nc state, north carolina, book, books, lessons for, lesson, education, children">
> ...

Meta Keywords are nowadays either ignored by search engines or may even increase the chances of the site being classified as spam. Other than that, your recommendations are very useful!

**Comment by Inlife-raphael** [comment]
Another factor for SEO is speed and security.

HTTP headers help security, they may help SEO. https://observatory.mozilla.org/analyze/landchad.net
NB: don't use HPKP as they advise, it's [dead](https://scotthelme.co.uk/hpkp-is-no-more/). We can use a very strict CSP policy. On Bloatpress, it'd break the website. But if you only get pictures from the server, you might allow only fetching images and styles for the domain (while JavaScript will be blocked wherever it is).

For speed, the website is already good (static + garbage-free). However, Free Software analyzers still have some remarks:
- https://pagespeed.web.dev/report?url=https%3A%2F%2Flandchad.net%2F 
- https://yellowlab.tools/result/g9r2fbgpjn
- https://validator.w3.org/nu/?doc=https%3A%2F%2Flandchad.net%2F

Little remark: absolutely all major search engines are proprietary, even cuckcuckgo or Brave Soyrch. Most tips are speculative, except the ones of the official search engines themselves:
- https://developers.google.com/search/docs/beginner/seo-starter-guide 
- https://www.bing.com/webmasters/help/webmasters-guidelines-30fba23a
- https://yandex.com/support/webmaster/controlling-robot/robots-txt.html

**Comment by binex-dsk** [comment]
> Another factor for SEO is speed

Good to know HTTPS is out of the question.
